Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -222,10 +222,57 @@ protected override void OnCreate(Bundle savedInstanceState)
 			base.OnCreate(savedInstanceState);
 
 			Profile.FramePartition("SetSupportActionBar");
-			AToolbar bar;
+			AToolbar bar = null;
+
+#if __ANDROID_29__
+			if (ToolbarResource == 0)
+			{
+				ToolbarResource = Resource.Layout.Toolbar;
+			}
+
+			if (TabLayoutResource == 0)
+			{
+				TabLayoutResource = Resource.Layout.Tabbar;
+			}
+#endif
+
 			if (ToolbarResource != 0)
 			{
-				bar = LayoutInflater.Inflate(ToolbarResource, null).JavaCast<AToolbar>();
+				try
+				{
+					bar = LayoutInflater.Inflate(ToolbarResource, null).JavaCast<AToolbar>();
+				}
+#if __ANDROID_29__
+				catch (global::Android.Views.InflateException ie)
+				{
+					if (ie.Cause is Java.Lang.ClassNotFoundException &&
+						ie.Message == "Binary XML file line #1: Error inflating class android.support.v7.widget.Toolbar" &&
+						this.TargetSdkVersion() >= 29)
+					{
+						Internals.Log.Warning(nameof(FormsAppCompatActivity), 
```

Comment:
If inflation fails and all the above criteria then output a message explaining how to upgrade the templates to Android X or just inform the user that they can delete the layouts all together